account/src/main/java/com/user/account/controller/AllControlers.java

```diff
@@ -1,65 +1,93 @@
 package com.user.account.controller;
 
+import java.util.ArrayList;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
 
 import com.user.account.entity.User;
-import com.user.account.services.AccountServices;
-
+import com.user.account.message.DefaultMessage;
+import com.user.account.entity.Transactions;
+import com.user.account.services.AccountService;
+import com.user.account.services.TransactionService;
 
+import javax.persistence.EntityNotFoundException;
+import javax.validation.constraints.Min;
 
+@Validated
 @RestController
+@RequestMapping("/api")
 public class AllControlers {
-	
-	
-		@Autowired
-		private AccountServices accountServices; 
-		
-		// tomcat test
-		@GetMapping("/home")
-		public String home() {
-			return "This is the home page";
-		}
-		
-		// Updating accountBalance after debit
-		
-		
-		@PostMapping("/debit/{accountNumber}/{amount}")
-		public User debit(@PathVariable String accountNumber, @PathVariable String amount) {
-			
-			User user = this.accountServices.getUser(Long.parseLong(accountNumber));
-			
-			if ( Long.parseLong(amount) <= user.getAccountBalance() ) {
-				
-				user.updateBalanceBy(-Long.parseLong(amount)) ;
-				
-				return this.accountServices.updateUser(user);
-			}
-			
-			else 
-				return user;
-		}
-		
-		// Updating accountBalance after credit
-		
-		@PostMapping("/credit/{accountNumber}/{amount}")
-		public User credit(@PathVariable String accountNumber, @PathVariable String amount) {
-				
-			User user = this.accountServices.getUser(Long.parseLong(accountNumber));
-			
-			if ( Long.parseLong(amount) <= user.getAccountBalance() ) {
-				
-				user.updateBalanceBy(Long.parseLong(amount)) ;
-				
-				return this.accountServices.updateUser(user);
-			}
-			
-			else 
-				return user;
-
-		}
+
+
+    @Autowired
+    private AccountService accountService;
+
+    @Autowired
+    private TransactionService transactionService;
+
+    public ResponseEntity<String> CreditAndDebitRequest(Long accountNumber, String transactionAmount, String transactionType){
+        try {
+            User user = accountService.getUser(accountNumber);
+            Long amount = Long.parseLong(transactionAmount);
```

Comment:
This function is too complicated, please discuss with @Abhinav-Cowkur and simpify it.

High level pointers:

DEBIT/CREDIT can be `enum`
amount is already checked to be more than 0
I still don't see @Transactional in below code
```
transactionService.addTransaction(newTransaction);
accountService.updateUser(user);
```

Reply:
Added Nitin. It will come in today's commit